### DISCUSSION TITLE
Autogenerated Screen for "complex" relationships does not work as expected (external PostgreSQL)

### DISCUSSION THREAD

**Discussion by Markus-** [Budibase/budibase#10398] [Help] [answered]
As stated in the other discussion I have a problem by using budibase for a more complex scenario.

My Screen:

![grafik](https://user-images.githubusercontent.com/9554640/234003658-b6dfb475-d22d-4eaa-b61e-8cb00d0b7c90.png)

My Details-Sidebar
![grafik](https://user-images.githubusercontent.com/9554640/234003855-3a74d36a-ca64-475f-95f9-e98f176e1683.png)

Relations:
products-table
products-table > approver (many-to-many through approver_product)
products-table > product_role (one-to-many)
products-table > service (many-to-many through stock)

I try to add an approver (+markus@test.de) and get the error "error: duplicate key value violates unique constraint "stock_pkey""
I try to remove an approver of product 4: Works
I try to add the approver again to product4: Works
I tryp to remove the approver from product2: error: duplicate key value violates unique constraint "stock_pkey"

So it seems to work on every entry which only got a relation to the approver-table, but all changes which are related to approver AND service simply do not work, because despite just changing the approver-part an INSERT into the stock-table is tried.

Any ideas?
Regards 
Markus

**Comment by melohagan** [comment]
Hey @Markus- 

I think this may be related to this bug here: https://github.com/Budibase/budibase/issues/10236

**Reply by Markus-** [reply]
So in general it should work, but it does not?

Could you reproduce the error on your side or how could I circumvent this issue?
Regards
Markus

**Comment by Markus-** [comment]
So I think it depends on the order of the relationship.

In addition to description above, I have the "same" approver part for the service.

So

products-table
products-table > approver (many-to-many through approver_product)
products-table > product_role (one-to-many)
service-table
service-table > products-table (many-to-many through stock)
service-table > approver (many-to-many through approver_service)

This was the definition BEFORE I dig deeper. 

The service table works as expected and I can change approver and products as I like.

Then I switched the relation from

service-table > products-table (many-to-many through stock) 

TO

product-table > service-table (many-to-many through stock)

Then it works in the products-table but not in the service-table anymore.
So the error is the same, but it switched places.

Does this help?

Regards
Markus

**Comment by melohagan** [comment, marked as answer]
Docs updated here: https://docs.budibase.com/docs/sql-datasource#many-to-many

Please ensure your joining table is configured to work with Budibase as described in that section.
